studio/components/interfaces/BillingV2/Subscription/Subscription.tsx

```diff
@@ -24,41 +23,13 @@ const Subscription = ({}: SubscriptionProps) => {
   const panel = router.query.panel
   const isOrgBilling = !!organization?.subscription_id
 
-  useEffect(() => {
-    if (isOrgBilling) {
-      const { ref, panel } = router.query
-      let redirectUri = `/org/${organization.slug}/billing`
-      switch (panel) {
-        case 'subscriptionPlan':
-          redirectUri = `/org/${organization.slug}/billing?panel=subscriptionPlan`
-          break
-        case 'costControl':
-          redirectUri = `/org/${organization.slug}/billing?panel=costControl`
-          break
-        case 'computeInstance':
-          redirectUri = `/project/${ref}/settings/addons?panel=computeInstance`
-          break
-        case 'pitr':
-          redirectUri = `/project/${ref}/settings/addons?panel=pitr`
-          break
-        case 'customDomain':
-          redirectUri = `/project/${ref}/settings/addons?panel=customDomain`
-          break
-      }
-
-      router.push(redirectUri)
-    }
-  }, [router, organization?.slug, isOrgBilling])
-
-  useEffect(() => {
-    if (panel && typeof panel === 'string' && allowedValues.includes(panel)) {
-      snap.setPanelKey(panel as SUBSCRIPTION_PANEL_KEYS)
-    }
-  }, [panel])
-
   // No need to bother rendering, we'll redirect anyway
   if (isOrgBilling) return null
 
+  if (panel && typeof panel === 'string' && allowedValues.includes(panel)) {
```

Comment:
just to check though - must this be outside of a useEffect?

Reply:
useEffect is probably safer for this, can adjust

Reply:
yeah if possible lets put this in a useEffect - i feel iffy about having this block re-running on each re-render

Reply:
Adjusted